storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogMetadataManager.java

```diff
@@ -201,4 +201,13 @@ void onPartitionLeadershipChanges(Set<TopicIdPartition> leaderPartitions,
      * @param partitions topic partitions that have been stopped.
      */
     void onStopPartitions(Set<TopicIdPartition> partitions);
+
+    /**
+     * Returns total size of the log for the given leader epoch in remote storage.
+     *
+     * @param topicPartition topic partition for which size needs to be calculated.
+     * @param leaderEpoch Size will only include segments belonging to this epoch.
+     * @return Total size of the log stored in remote storage in bytes.
+     */
+    Long remoteLogSize(TopicIdPartition topicPartition, int leaderEpoch) throws RemoteStorageException;
```

Comment:
What is the reason for using `Long` instead of `long` and when can it be null? I prefer this to be primitive type unless there is a strong reason for the wrapper object.

Reply:
Rename the variable from `topicPartition` -> `topicIdPartition` for uniformity.

Reply:
Thank you for the reviews! @satishd to be honest this is a miss on my side as I started with the interface specified here https://cwiki.apache.org/confluence/display/KAFKA/KIP-852%3A+Optimize+calculation+of+size+for+log+in+remote+tier#KIP852:Optimizecalculationofsizeforloginremotetier-NewAPIinRemoteLogMetadataManager(RLMM). Looking over it again, in its current implementation the value can never be null as if we cannot calculate the size due to a storage problem we will throw an exception. I will change it in subsequent commits + update the KIP.

Reply:
+1 to change it in subsequent commits + update the KIP.